code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/internal/InAppNotificationHandler.java

```diff
@@ -202,19 +200,13 @@ private void processPropositions(final List<PropositionPayload> propositions, fi
         final Map<String, PropositionInfo> tempPropositionInfo = new HashMap<>();
 
         for (final PropositionPayload proposition : propositions) {
-            if (proposition != null && proposition.propositionInfo != null && !expectedScope.equals(proposition.propositionInfo.scope)) {
-                Log.debug(MessagingConstants.LOG_TAG, SELF_TAG, "processPropositions - Ignoring proposition where scope (%s) does not match expected scope (%s).", proposition.propositionInfo.scope, expectedScope);
-                continue;
-            }
-
             if (proposition == null) {
-                Log.trace(LOG_TAG, SELF_TAG, "processPropositions - Processing aborted, null proposition found.");
-                return;
+                continue;
             }
```

Comment:
is this check still necessary with the new handling of `PropositionPayload` creation?